dataactcore/read_config.py

```diff
@@ -0,0 +1,165 @@
+
+VALUES_TO_LOG = {"CONFIG_BROKER":["path","use_aws","local",'aws_bucket', 'aws_role', 'aws_region', 'aws_create_temp_credentials',
+                                  "static_files_bucket", "help_files_path", "d_file_storage_path", "broker_files","award_procurement_url",
+                                  "award_procurement_file_name", "award_url", "award_file_name", "awardee_attributes_url",
+                                  "awardee_attributes_file_name", "sub_award_url", "sub_award_file_name", "rss_folder",
+                                  "rss_file", "generated_files_folder", "sf_133_folder", "sf_133_file"],
+                 "CONFIG_SERVICES":["error_report_path","broker_api_host","validator_host","broker_api_port","validator_port",
+                                    "protocol", "rest_trace", "server_debug"],
+                 "CONFIG_DB":["dynamo_host","dynamo_port","base_db_name","scheme"],
+                 "CONFIG_LOGGING":["log_files","use_logstash","logstash_host","logstash_port"],
+                 "CONFIG_JOB_QUEUE":["url", "port", "broker_scheme"]}
+
+# set the location of the DATA Act broker config files
+CONFIG_PATH = os.path.join(dirname(abspath(__file__)), 'config.yml')
+if "env" in os.environ:
+    env = os.environ["env"]
+    ENV_PATH = os.path.join(dirname(abspath(__file__)), '{}_config.yml'.format(env))
+    SECRET_PATH = os.path.join(dirname(abspath(__file__)), '{}_secrets.yml'.format(env))
+    path_list = [CONFIG_PATH, ENV_PATH, SECRET_PATH]
+else:
+    # If no environment set, just use the base config
+    path_list = [CONFIG_PATH]
+# set the location of the Alembic config file
+ALEMBIC_PATH = os.path.join(dirname(abspath(__file__)), 'alembic.ini')
+MIGRATION_PATH = os.path.join(dirname(abspath(__file__)), 'migrations')
+CONFIG_BROKER = {}
+CONFIG_SERVICES = {}
+CONFIG_DB = {}
+CONFIG_LOGGING = {}
+CONFIG_JOB_QUEUE = {}
+
+def merge_config(existing_config, new_config):
+    if type(existing_config) is not dict or type(new_config) is not dict:
+        raise ValueError("Both args to merge_config must be dicts")
+    for key in new_config.keys():
+        existing_config[key] = new_config[key]
+    return existing_config
+
+first_config = True
+log_message = ""
+for config_path in path_list:
+    try:
+        with open(config_path) as c:
+            CONFIG_ALL = yaml.load(c)
+    except IOError:
+        if first_config:
+            # First config file must be present, the rest are optional
+            raise IOError('Error reading the config file. Please make sure this file exists'
+               ' before starting the DATA Act broker: {}'.format(config_path))
+        else:
+            log_message = " ".join([log_message, "### Did not load config file from {}".format(config_path), "warning"])
+            continue
+    # File successfully loaded, mark as no longer first
+    first_config = False
+
+    CONFIG_BROKER = merge_config(CONFIG_BROKER, CONFIG_ALL['broker']) if "broker" in CONFIG_ALL else CONFIG_BROKER
```

Comment:
Switched to using update and did some related simplifications, merge_config is now gone